webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/reassignHostGroup/actions.js

```diff
@@ -0,0 +1,27 @@
+import { APIActions } from '../../../../redux/API';
+import { foremanUrl } from '../../../../common/helpers';
+
+export const BULK_REASSIGN_HOSTGROUP_KEY = 'BULK_REASSIGN_HOSTGROUP_KEY';
+export const bulkReassignHostgroups = (params, handleSuccess, handleError) => {
+  const url = foremanUrl(`/api/v2/hosts/bulk/reassign_hostgroup`);
+  return APIActions.put({
+    key: BULK_REASSIGN_HOSTGROUP_KEY,
+    url,
+    successToast: response => response.data.message,
+    handleSuccess,
+    handleError,
+    params,
+  });
+};
+
+export const HOSTGROUP_KEY = 'HOSTGROUP_KEY';
+
+export const fetchHostgroups = () => {
```

Comment:
How does this deal with pagination? I don't see it passing `per_page=all` or having any way to select the next page.

Reply:
Hmm..There is no pagination to support the dropdown so I guess we need to send the per_page=all to support large number of hostgroups.
![Hostgroup_sketch](https://github.com/theforeman/foreman/assets/21146741/3f243dd8-fac9-47af-8341-a939a4b92046)

Reply:
Updated to fetch all Hostgroups for dropdown.